File: app/src/main/java/com/botijasoftware/utils/Viewport.java
package com.botijasoftware.utils;

import javax.microedition.khronos.opengles.GL10;
import com.botijasoftware.utils.renderer.Renderer;

import android.opengl.GLES20;

public class Viewport {
	private int X, Y, width, height;

	public Viewport(int x, int y, int w, int h) {
		X = x;
		Y = y;
		width = w;
		height = h;
	}
	
	public void enable(GL10 gl) {
		GLES20.glViewport(X, Y, width, height);
		//GLES20.glLoadIdentity();
		Renderer.modelview.loadIdentity();
	}
	
	public void getAsArray(int[] viewport) {
		viewport[0] = X;
		viewport[1] = Y;
		viewport[2] = width;
		viewport[3] = height;
	}
}

File: app/src/main/java/com/botijasoftware/utils/Viewport.kt
package com.botijasoftware.utils

import com.botijasoftware.utils.renderer.Renderer

import android.opengl.GLES20

class Viewport(private val X: Int, private val Y: Int, private val width: Int, private val height: Int) {

    fun enable() {
        GLES20.glViewport(X, Y, width, height)
        //GLES20.glLoadIdentity();
        Renderer.modelview.loadIdentity()
    }

    fun getAsArray(viewport: IntArray) {
        viewport[0] = X
        viewport[1] = Y
        viewport[2] = width
        viewport[3] = height
    }
}
